elements/dash-checkbox/dash-checkbox.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * // TODO
+ */
+(function() {
+  /**
+   * Define key codes to help with handling keyboard events.
+   */
+  const KEYCODE = {
+    SPACE: 32,
+  };
+
+  class DashCheckbox extends HTMLElement {
+    static get observedAttributes() {
+      return ['checked'];
+    }
+
+    constructor() {
+      super();
+    }
+
+    connectedCallback() {
+      this.setAttribute('role', 'checkbox');
+      if (!this.hasAttribute('aria-checked'))
+        this.setAttribute('aria-checked', 'false');
+      if (!this.hasAttribute('tabindex'))
+        this.setAttribute('tabindex', 0);
+
+      this.addEventListener('keydown', this._onKeyDown);
+      this.addEventListener('click', this._onClick);
+    }
+
+    _onKeyDown(event) {
+      // Don’t handle modifier shortcuts typically used by assistive technology.
+      if (event.altKey) return;
+
+      switch (event.keyCode) {
+        case KEYCODE.SPACE:
+          this._toggleChecked();
+          break;
+        // Any other key press is ignored and passed back to the browser.
+        default:
+          return;
+      }
+    }
+
+    _onClick(event) {
+      this._toggleChecked();
+    }
+
+    _toggleChecked() {
+      const newState = !this.checked;
+      this.checked = newState;
+      this.dispatchEvent(new CustomEvent('check-changed', {
```

Comment:
Wouldn’t it be good to make this compatible to a normal `input[type=checkbox]` and fire a `change` event that bubbles?